### ISSUE TITLE
Consistent URI_PARAMS implict dispatcher behavior

### ISSUE THREAD

**Issue by rquinio1A** [microcks/microcks#1132]
### Reason/Context

As as a Microcks newbe, I was trying to mock the API https://github.com/amadeus4dev/amadeus-open-api-specification/blob/main/spec/yaml/HotelList_v1_swagger_specification.yaml#L98 (which is not Microcks-friendly!), and I've been a bit surprised by the behavior of the implict URI_PARAMS

For instance endpoint http://localhost:80/rest/Hotel+List/1.0.4/reference-data/locations/hotels/by-city?cityCode=NCE
has two response examples: "Successful Reply" and "Invalid Format" and there's no matching request parameter examples. 
The behavior of the dispatcher is that it decides at startup to answer either one or the other response example consitently - but which example is selected isn't consitent across Microcks restarts, making the testing flaky.

This can be solved by properly adding missing examples in the API spec or customizing the dispatch with a secondary artifact, but I was wondering if the initial dev experience could be better.

### Description

If the request is ambiguous, meaning the dispatcher could theorically dispatch to more than 1 example:
* always default to the first (in the document order ?) - better initial experience but harder for the user to troubleshoot.
* or fail with a 404 error with a description of the ambiguity

Also parameters without examples in a request should not invalidate a partial matching. 

So if the spec had:

```yaml
parameters:
  - name: cityCode
     examples:
         Successful Reply:
            value: NCE
         Invalid Format:
            value: Paris
  - name: ratings
    # no examples for ratings
```

Then: 

* http://localhost:80/rest/Hotel+List/1.0.4/reference-data/locations/hotels/by-city?cityCode=NCE => Successful Reply
* http://localhost:80/rest/Hotel+List/1.0.4/reference-data/locations/hotels/by-city?cityCode=Paris => Invalid Format
* http://localhost:80/rest/Hotel+List/1.0.4/reference-data/locations/hotels/by-city?cityCode=PAR => Successful Reply or 404 (ambiguous case)
* http://localhost:80/rest/Hotel+List/1.0.4/reference-data/locations/hotels/by-city?cityCode=NCE&ratings=5 =>  Successful Reply (current behavior is to return either "Successful Reply" or "Invalid Format")

**Comment by lbroudoux** [comment]
Hello @rquinio1A,

I'm a bit surprised as this clunky behavior should have been fixed with https://github.com/microcks/microcks/issues/819#issuecomment-1984487140 
I will have a look at it.

**Comment by lbroudoux** [comment]
You're right - it has not been fully implemented into the `RestController`. The implementation is completed for SOAP for not for REST.
However, we don't have implemented that using a `404` return code for SOAP but rather a `400` with an error message saying that `The response with dispatch criteria 'XYZ' cannot be found`. At that time, we thought it was better to say that the client request was malformed and could not be processed by the server than to return a `404`, which is commonly defined in REST APIs. So that we have more chances to make the difference between a processing error and an expected one that mimics the behavior of the application... Any thoughts on this?

**Comment by lbroudoux** [comment]
This is a follow-up of the issue #819

**Comment by rquinio1A** [comment]
Great! I haven't tried SOAP yet, so I didn't notice the discrepancy.
Seeing the discussion on 404 vs 400, I understand the arguments for `400` indeed, `404` rather makes sense when the endpoint path http://localhost:80/rest/Hotel+List/1.0.4/ wouldn't exist at all because of wrong spec name or version.
